### ISSUE TITLE
Case Processor?

### ISSUE THREAD

**Issue by loneprairie** [mecachisenros/caldera-forms-civicrm#42]
Hi again,

Could this possibly be used to open a new case in CiviCase?

Thanks

**Comment by mecachisenros** [comment]
@loneprairie it could be used, although is not implemented yet.

Design wise I need to figure out if a _Case Processor_ is needed or will it be sufficient with incorporating the functionality in the Activity processor, ie having a _File Activity to Case_ checkbox which would show Case related options when clicked.

Also think about more functionality like auto-fill form based activity id etc.

So any input, thoughts are appreciated.

**Comment by loneprairie** [comment]
@mecachisenros  I see - here is the envisioned workflow:

1. A visitor to our website is in need of assistance with a new issue and fills out our "Get Help" form.
2. Upon submitting the form, Civi creates a new contact or updates and existing one as necessary and opens a new case for that contact
3. Our staff then have a case and case timeline set up for them automatically as a result of the user submitting the form

In other words, the cases do not exist until the user contacts us. So, _File Activity to Case_ would not help in this scenario. Currently, we have to manually create the case after this form is submitted.

Hopefully, this helps explain our use.

Thanks again for your work on this!

**Comment by mecachisenros** [comment]
@loneprairie I've created a new branch here https://github.com/mecachisenros/caldera-forms-civicrm/tree/civicase with a Case processor (basic implementation) if you could test and give feedback it would be highly appreciated.

**Comment by loneprairie** [comment]
@mecachisenros Thank you! I installed without incident and the new processor is available to me. Currently, I cannot get the processor to create a new case and I suspect it has to do with the mapping of a couple of fields required by CiviCase when creating a new case:

1. Activity Medium - this should be mapped to the Activity Medium field in the New Case Profile (I am not sure if this is actually a profile since I cannot find it in the Profile lists). At any rate, if you go to Civi>Cases>New Case, you will see the field in question.
2. Subject - This a text field in the New Case Profile. Like Activity Medium, this can be seen at Civi>Cases>New Case. Can this be filled by using a hidden field in the Caldera Form that contains a predefined text string?

Thanks again

**Comment by mecachisenros** [comment]
@loneprairie I believe Activity Medium is not required when using the API (it will default to _email_), but you are correct Subject it is required. You can map a hidden field for the Subject or set up a dropdown field with different options so the user can chose from, or map it to a text filed and require it in the form

**Comment by loneprairie** [comment]
@mecachisenros You are correct and the processor does now create a case - well done! There is one more feature that would be handy, if possible: currently, the Civi sets the Case Role of Case Coordinator (Case Manager) as the user who submits the form but that role should actually be assigned to a staff member/existing Civi user. Is it possible to map a specific Civi user into that role using the case processor?

Thanks for your work on this - it will be so helpful for us!

**Comment by mecachisenros** [comment]
@loneprairie the [latest release](https://github.com/mecachisenros/caldera-forms-civicrm/releases) and master should fix this.

**Comment by JVeitia** [comment]
Thank you for this development, I'm finding it very useful in my organization's workflows/roles.
